src/Analyzer.JsonRuleEngine.FunctionalTests/ScopeSelectionTests.cs

```diff
@@ -0,0 +1,254 @@
+        ""Tag2"": ""AnotherTag""
+      },
+      ""dependsOn"": [
+        ""[concat('Microsoft.Network/networkInterfaces/', 'myNIC2')]""
+      ],
+      ""properties"": {
+        ""osProfile"": {
+          ""computerName"": ""vm2"",
+          ""customData"": ""some data""
+        },
+        ""networkProfile"": {
+          ""networkInterfaces"": [
+            {
+              ""id"": ""[resourceId('Microsoft.Network/networkInterfaces', 'myNIC2')]""
+            }
+          ]
+        }
+      }
+    }
+  ]
+}";
+        #endregion
+
+        /// <summary>
+        /// A mock implementation of an <see cref="Expression"/> for testing internal methods.
+        /// </summary>
+        private class MockExpression : Expression
+        {
+            public Action<IJsonPathResolver> EvaluationCallback { get; set; }
+
+            public MockExpression(ExpressionCommonProperties commonProperties)
+                : base(commonProperties)
+            { }
+
+            public override JsonRuleEvaluation Evaluate(IJsonPathResolver jsonScope)
+            {
+                return base.EvaluateInternal(jsonScope, scope =>
+                {
+                    EvaluationCallback(scope);
+                    return new JsonRuleResult();
+                });
+            }
+        }
+
+        [DataTestMethod]
+        [DataRow(null, "outputs",
+            null,
+            DisplayName = "No resource type, path not resolved")]
+        [DataRow(null, "$schema",
+            "$schema",
+            DisplayName = "No resource type, path resolved")]
+        [DataRow(null, "params.*",
+            // No scopes expected evaluated
+            DisplayName = "No resource type, wildcard path does not resolve")]
+        [DataRow(null, "parameters.*",
+            "parameters.location",
+            DisplayName = "No resource type, wildcard path resolves single path")]
+        [DataRow(null, "resources[*]",
+            "resources[0]", "resources[1]", "resources[2]", "resources[3]", "resources[4]",
+            DisplayName = "No resource type, wildcard path resolves multiple paths")]
+        [DataRow("Microsoft.Storage/storageAccounts", null,
+            // No scopes expected evaluated
+            DisplayName = "Resource type matches none, no path")]
+        [DataRow("Microsoft.Network/virtualNetworks", null,
+            "resources[0]",
+            DisplayName = "Resource type matches 1, no path")]
+        [DataRow("Microsoft.Compute/virtualMachines", null,
+            "resources[3]", "resources[4]",
+            DisplayName = "Resource type matches multiple, no path")]
+        [DataRow("Microsoft.Storage/storageAccounts", "name",
+            // No scopes expected evaluated
+            DisplayName = "Resource type matches none, path not resolved")]
+        [DataRow("Microsoft.Network/virtualNetworks", "properties.addressSpace",
+            null,
+            DisplayName = "Resource type matches 1, path not resolved")]
```

Comment:
is null not the same thing as not passing in a parameter/ why did you decide not (also line 214)?

Reply:
Maybe I should update the comments.  If a static path isn't found, null is evaluated as the path value (since it isn't found).  If a wildcard path isn't found, it returns empty, and nothing is evaluated.